agent/pkg/crane/k8s/deployment.go

```diff
@@ -81,7 +81,7 @@ func (d *deployment) deployDeployment(p *deploymentParams) error {
 					d.appConfig.KeyIssuer:  p.issuer,
 				}).WithSpec(
 					corev1.PodSpec().WithContainers(containerConfig).
-						WithInitContainers(getInitContainers(p.containerConfig, d.appConfig)...).
+						WithInitContainers(getInitContainers(p.containerConfig, d.appConfig, p)...).
```

Comment:
If `p` is passed to this function then passing `p.containerConfig` could be simplyfied just by passing `p` and then getting `containerConfig` later in the function.